hdfs/client.py

```diff
@@ -477,7 +477,10 @@ def consumer(_data):
         raise _to_error(res)
 
     if data is None:
-      return AsyncWriter(consumer)
+      if buffersize is None:
+        return AsyncWriter(consumer)
+      else:
+        return BoundedAsyncWriter(consumer, buffer_size=buffersize)
```

Comment:
`buffersize` is an HDFS argument, better not to overload its use here. Let's introduce a separate argument instead.